dynamic_instrumentation/src/into_operand.rs

```diff
@@ -0,0 +1,42 @@
+use rustc_middle::{
+    mir::{Constant, Local, Operand, Place},
+    ty::{self, ParamEnv, TyCtxt},
+};
+use rustc_span::DUMMY_SP;
+
+/// Types that can be passed as an argument to instrumentation hook functions.
+pub trait IntoOperand<'tcx> {
+    fn op(self, tcx: TyCtxt<'tcx>) -> Operand<'tcx>;
+}
+
+impl<'tcx> IntoOperand<'tcx> for Place<'tcx> {
+    fn op(self, _tcx: TyCtxt) -> Operand<'tcx> {
+        Operand::Copy(self)
+    }
+}
+
+impl<'tcx> IntoOperand<'tcx> for u32 {
+    fn op(self, tcx: TyCtxt<'tcx>) -> Operand<'tcx> {
+        make_const(tcx, self)
+    }
+}
+
+impl<'tcx> IntoOperand<'tcx> for Local {
+    fn op(self, tcx: TyCtxt<'tcx>) -> Operand<'tcx> {
+        Place::from(self).op(tcx)
+    }
+}
+
+impl<'tcx> IntoOperand<'tcx> for Operand<'tcx> {
+    fn op(self, _tcx: TyCtxt<'tcx>) -> Self {
+        self
+    }
+}
+
+fn make_const(tcx: TyCtxt, idx: u32) -> Operand {
```

Comment:
i think functions like these and most of the functions in `deref` should be combined into a `mir_utils` module

Reply:
This can be encapsulated by this module and made private, which is why I put it here.

Reply:
no need for it to be private, it can be useful in future contexts

Reply:
Unlike `rv_place`, I don't agree.  The `IntoOperand` `impl` does exactly the same, and I think it's bad to have two exposed functions that do the exact same thing, which causes people to end up using both randomly.

Reply:
ok, on second thought I think you're right and `into_operand` can stay its own module